packages/app-layout/src/vaadin-app-layout.js

```diff
@@ -548,6 +568,58 @@ class AppLayout extends ElementMixin(ThemableMixin(mixinBehaviors([IronResizable
     // TODO(jouni): ARIA attributes. The drawer should act similar to a modal dialog when in ”overlay” mode
   }
 
+  /**
+   * Returns a promise that resolves when the drawer opening/closing CSS transition ends.
+   *
+   * The method relies on the `--vaadin-app-layout-transition` CSS variable to detect whether
+   * the drawer has a CSS transition that needs to be awaited. If the CSS variable equals `none`,
+   * the promise resolves immediately.
+   *
+   * @return {Promise}
+   * @private
+   */
+  __drawerTransitionComplete() {
+    return new Promise((resolve) => {
+      if (getComputedStyle(this).getPropertyValue('--vaadin-app-layout-transition').trim() === 'none') {
+        resolve();
+        return;
+      }
+
+      this.$.drawer.addEventListener('transitionend', resolve, { once: true });
+    });
+  }
+
+  /** @private */
+  __trapFocusInDrawer() {
+    this.$.drawer.setAttribute('tabindex', '0');
```

Comment:
Is it necessary to make the drawer focusable? The focus outline looks a bit overwhelming

<img width="576" alt="Screenshot 2021-12-09 at 16 31 51" src="https://user-images.githubusercontent.com/1222264/145415688-a92c0ee5-a664-4536-923a-4d08c32c1550.png">

Reply:
Agreed. We should not let the default outline appear, but rather use CSS in Lumo instead.

Reply:
wdyt @jouni ?

Reply:
I wonder what will happen in the case when there are no focusable elements in the drawer? At the moment, the drawer having `tabindex=0` guarantees that in such a case (1) the `dialog` role will be announced (2) `FocusTrapController` will not throw the corresponding `Please, provide at least one focusable element` exception. Also, if there are no focusable elements, it is not possible to close such an overlay with Escape. 

Although it sounds like an edge-case, I wonder if it would be reasonable to warn the user in some way?

Reply:
Maybe it could work the same way as dialog (= the drawer gets focused but it's not visually indicated)? https://vaadin.com/docs/latest/ds/components/dialog

Reply:
> Maybe it could work the same way as dialog (= the drawer gets focused but it's not visually indicated)?

The idea to get it in line with the dialog sounds good to me. If no objections from @jouni, I would go by this way.

Reply:
Sounds good to me.

Reply:
Removed the focus outline by adding `outline: none` to the drawer.